### PR TITLE
Remove manageBoards Tour Point

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
Removes the ManageBoards Tour Point.

Fixing this issue was more than a quick fix. Based on the way we load the sidebar, it's not easy to tell which is the first visible board (cannot simply use index 0, since board may be hidden. Also these are loaded looping through "all boards", which means index(0) is the first board alphabetically.

I tried several quick fixes, but nothing worked correctly. Talked to @wuwinson and decided to pull the tour point vs spending more time on it.

#### Ticket Link
  Fixes https://github.com/mattermost/focalboard/issues/3669

### CODE REVIEW

**Comment by Pinjasaur** [comment]
@sbishel confirming we're looking to remove it on `main` _and_ cherrypick to `release-7.2` too? wasn't sure if it could be removed in 7.2 and fixed for 7.3? 0/5

**Comment by wuwinson** [comment]
We should try to fix this in `v7.3` if possible, but don't want to hold up the `v7.2` release over this.

**Comment by sbishel** [comment]
> confirming we're looking to remove it on main and cherrypick to release-7.2 too? wasn't sure if it could be removed in 7.2 and fixed for 7.3? 0/5

Yes, if we want to reimplement for 7.3, we can "revert" this, but I don't want to leave dead code in there.

File: webapp/src/components/shareBoard/teamPermissionsRow.tsx
Comment (Pinjasaur) [inline webapp/src/components/shareBoard/teamPermissionsRow.tsx:94]: We don't need this conditional—it's why the snapshot tests are failing. It only needed to be fixed in `release-7.2` as I did earlier in #3681.

**Review by Pinjasaur — APPROVED** [review]
@sbishel one note, otherwise it LGTM once that's fixed & CI is happy 🎉

File: webapp/src/components/shareBoard/teamPermissionsRow.tsx
Comment (sbishel) [reply webapp/src/components/shareBoard/teamPermissionsRow.tsx:94]: Yep...bad merge. This change shouldn't be here.

**Pull request merged**
